pymongo/pool.py

```diff
@@ -1557,7 +1571,9 @@ def remove_stale_sockets(self, reference_generation: int) -> None:
                     if self.gen.get_overall() != reference_generation:
                         conn.close_conn(ConnectionClosedReason.STALE)
                         return
-                    self.conns.appendleft(conn)
+                    self.available_conns.appendleft(conn)
+                    if conn.cancel_context in self.active_conns_context:
+                        self.active_conns_context.remove(conn.cancel_context)
```

Comment:
set.remove already handles this case so the if statement can be removed.

Reply:
Turns out `set.discard` is the method we're looking for, good catch!

Reply:
Ah that's the one!